`./x.py doc` on a clean repo on master passes, but with warnings about feature gates

```rust
[...]
Documenting error index (x86_64-unknown-linux-gnu)
Building stage0 tool error_index_generator (x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.18s
Generating lint docs (x86_64-unknown-linux-gnu)
Building stage0 tool lint-docs (x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.15s
warning: the code example in lint `unfulfilled_lint_expectations` in /home/jess/src/rust/compiler/rustc_lint_defs/src/builtin.rs failed to generate the expected output: did not find lint `unfulfilled_lint_expectations` in output of example, got:

error[E0554]: `#![feature]` may not be used on the beta release channel
 --> lint_example.rs:1:1
  |
1 | #![feature(lint_reasons)]
  | ^^^^^^^^^^^^^^^^^^^^^^^^^


error: aborting due to previous error


For more information about this error, try `rustc --explain E0554`.

warning: the code example in lint `unused_allocation` in /home/jess/src/rust/compiler/rustc_lint/src/unused.rs failed to generate the expected output: did not find lint `unused_allocation` in output of example, got:

error[E0554]: `#![feature]` may not be used on the beta release channel
 --> lint_example.rs:1:1
  |
1 | #![feature(box_syntax)]
  | ^^^^^^^^^^^^^^^^^^^^^^^


error: aborting due to previous error


For more information about this error, try `rustc --explain E0554`.

Rustbook (x86_64-unknown-linux-gnu) - rustc
Documenting stage0 clippy (x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.21s
Documenting stage0 miri (x86_64-unknown-linux-gnu)
    Finished release [optimized] target(s) in 0.17s
Build completed successfully in 0:01:10
./x.py doc  63.71s user 11.62s system 105% cpu 1:11.48 total
```

This should probably fail docs, but also it shouldn't happen in the first place. Not entirely sure why the other lines in the file don't trigger this, but I'm just making a note of this.

--- a/src/bootstrap/doc.rs
+++ b/src/bootstrap/doc.rs
@@ -1027,10 +1027,11 @@ impl Step for RustcBook {
         if self.validate {
             cmd.arg("--validate");
         }
-        if !builder.unstable_features() {
-            // We need to validate nightly features, even on the stable channel.
-            cmd.env("RUSTC_BOOTSTRAP", "1");
-        }
+        // We need to validate nightly features, even on the stable channel.
+        // Set this unconditionally as the stage0 compiler may be being used to
+        // document.
+        cmd.env("RUSTC_BOOTSTRAP", "1");
+
         // If the lib directories are in an unusual location (changed in
         // config.toml), then this needs to explicitly update the dylib search
         // path.

